src/ui/components/results/paginationcomponent.js

```diff
@@ -113,13 +113,17 @@ export default class PaginationComponent extends Component {
      */
     this._limit = this.core.storage.get(StorageKeys.SEARCH_CONFIG).limit;
 
-    const offset = this.core.globalStorage.getState(StorageKeys.SEARCH_OFFSET) || 0;
-    this.core.globalStorage.set(StorageKeys.SEARCH_OFFSET, Number(offset));
-    this.core.globalStorage.on('update', StorageKeys.SEARCH_OFFSET, offset => {
-      if (typeof offset === 'number') {
-        return;
+    const offset = this.core.storage.get(StorageKeys.SEARCH_OFFSET) || 0;
+    this.core.storage.set(StorageKeys.SEARCH_OFFSET, Number(offset));
+    this.core.storage.registerListener({
+      eventType: 'update',
+      storageKey: StorageKeys.SEARCH_OFFSET,
+      callback: offset => {
```

Comment:
This feels like it's not pagination component's job (also not part of this PR, just commenting)

Reply:
🍝 🍝 🍝